web-registry/src/types/ledger/ecocredit.ts

```diff
@@ -73,6 +73,73 @@ export interface QueryBasketsResponse {
   pagination?: PageResponse;
 }
 
+export interface IBasket {
+  id: string;
+  basketDenom: string;
+  name: string;
+  disableAutoRetire: boolean;
+  creditTypeAbbrev: string;
+  dateCriteria: {
+    minStartDate?: string;
+    startDateWindow?: string;
+  };
+  exponent: string;
+}
+
+export interface QueryBasketRequest {
+  basketDenom: string;
+}
+
+export interface QueryBasketResponse {
+  basket?: IBasket;
+  classes: string[];
+}
+
+export interface QueryBasketBalancesRequest {
+  basketDenom: string;
+  // pagination?: PageRequest; // DISABLED
+}
+
+export interface QueryBasketBalancesResponse {
+  balances: BasketBalance[];
+  // pagination?: PageResponse; // DISABLED
+}
+
+// BasketBalance stores the amount of credits from a batch in a basket
+export interface BasketBalance {
+  basketId: string | Long;
+  batchDenom: string;
+  balance: string;
+  batchStartDate?: string | Date;
+}
+
+/**
+ * Ecocredits
+ */
+
+/** BatchInfo represents the high-level on-chain information for a credit batch. */
+export interface IBatchInfo {
```

Comment:
I'd rather get rid of the `I` too
plus the interface names are not even consistent anymore, some have `I` while some other don't, which is confusing

Reply:
ok, as I was commenting, it was a forced renaming by the interfaces that have entered with the new changes, when updating my branch. The interfaces I'm using already have properties in camelCase, but the ones that entered master first go in snake_case. So they are not directly reusable. I just did a trivial rename because I can't reuse those interfaces that have come in without modifying my code or without modifying the existing one. And if we define 2 interfaces with the same name the interface is extended. What should I do?

Reply:
oh ok I get it now, this is because the responses from the REST endpoints come with snake_case while the response from regen-js will comme with camelCase
I think that we should just keep using the REST endpoint for now for querying batch info, so let's just use the existing snake_case `BatchInfo` interface everywhere (and we'll change that later on as part of https://github.com/regen-network/regen-registry/issues/852
Then when it comes to basket, since we haven't implemented any actual queries yet for those, let's just use the camelCase based interfaces everywhere since we'll start using regen-js for querying those.
wdyt?

Reply:
Now I just fully understood it too.. I leave the BatchInfo REST version and update Basket regen-js version. Wonderful Marie, thanks!